addons/project/models/project_task.py

```diff
@@ -640,6 +642,11 @@ def _compute_display_parent_task_button(self):
         for task in self:
             task.display_parent_task_button = task.parent_id in accessible_parent_tasks
 
+    def _compute_current_user_same_company_partner(self):
+        current_user = self.env.user
+        for task in self:
+            task.current_user_same_company_partner = task.partner_id and current_user.partner_id.commercial_partner_id == task.partner_id.commercial_partner_id
```

Comment:
We could directly store in the variable the commercial_partner_id of the current user. 🙂 

```suggestion
        commercial_partner_id = self.env.user.partner_id.commercial_partner_id
        for task in self:
            task.current_user_same_company_partner = task.partner_id and commercial_partner_id == task.partner_id.commercial_partner_id
```